migrator/models/cdn.py

```diff
@@ -0,0 +1,187 @@
+
+
+class CdnUserData(CdnModel):
+    """
+    CdnUserData is about the Let's Encrypt user associated with a CdnRoute.
+    We probably have no reason to ever think about this model in this project.
+    """
+
+    __tablename__ = "user_data"
+
+    id = sa.Column(sa.Integer, primary_key=True)
+    created_at = sa.Column(postgresql.TIMESTAMP)
+    updated_at = sa.Column(postgresql.TIMESTAMP)
+    deleted_at = sa.Column(postgresql.TIMESTAMP, index=True)
+    email = sa.Column(sa.Text, nullable=False)
+    reg = sa.Column(postgresql.BYTEA)
+    key = sa.Column(postgresql.BYTEA)
+
+
+class CdnRoute(CdnModel, RouteModel):
+    """
+    CdnRoute represents the core of the service instance
+    """
+
+    __tablename__ = "routes"
+
+    id = sa.Column(sa.Integer, primary_key=True)
+    # domain_internal is effectively the domain name of the CloudFront distribution
+    domain_internal = sa.Column(sa.Text)
+    # domain_external is a comma-separated list of domain names the user wants the CloudFront distribution to respond to
+    domain_external = sa.Column(sa.Text)
+    created_at = sa.Column(postgresql.TIMESTAMP)
+    updated_at = sa.Column(postgresql.TIMESTAMP)
+    deleted_at = sa.Column(postgresql.TIMESTAMP, index=True)
+    instance_id = sa.Column(sa.Text, index=True, nullable=False)
+    dist_id = sa.Column(sa.Text)
+    origin = sa.Column(sa.Text)
+    path = sa.Column(sa.Text)
+    insecure_origin = sa.Column(sa.Boolean)
+    challenge_json = sa.Column(postgresql.BYTEA)
+    user_data_id = sa.Column(sa.Integer, sa.ForeignKey("acme_user_v2.id"))
+    certificates: List["CdnCertificate"] = orm.relationship(
+        "CdnCertificate",
+        order_by="desc(CdnCertificate.expires)",
+        primaryjoin="(foreign(CdnCertificate.route_id)) == CdnRoute.id",
+        backref="route",
+    )
+    # state should be one of:
+    # deprovisioned
+    # provisioning
+    # provisioned
+    state = sa.Column(sa.Text, index=True, nullable=False)
+    operations: List["CdnOperation"] = orm.relationship(
+        "CdnOperation", backref="route", lazy="dynamic"
+    )
+    acme_user_id = sa.Column(sa.Integer, sa.ForeignKey("acme_user_v2.id"))
+    route_type = RouteType.CDN
+
+    def domain_external_list(self):
+        return self.domain_external.split(",")
+
+
+class CdnCertificate(CdnModel, CertificateModel):
+    __tablename__ = "certificates"
+
+    id = sa.Column(sa.Integer, primary_key=True)
+    created_at = sa.Column(postgresql.TIMESTAMP)
+    updated_at = sa.Column(postgresql.TIMESTAMP)
+    deleted_at = sa.Column(postgresql.TIMESTAMP, index=True)
+    route_id = sa.Column(sa.Integer)
+    domain = sa.Column(sa.Text)
+    # cert_url is the Let's Encrypt URL for the certificate
+    cert_url = sa.Column(sa.Text)
+    # certificate is the actual body of the certificate chain
+    # this was used by the old broker, but the renewer uses fullchain_pem and leaf_pem instead
+    certificate = sa.Column(postgresql.BYTEA)
```

Comment:
This is kept here for backward compatibility purposes?

Reply:
basically, yes. I kept it on the legacy-domain-certificate-renewer so alembic wouldn't try to drop the columns. I left it here so it's easier to sync if/when the models change again